src/interface/boxplot.jl

```diff
@@ -0,0 +1,168 @@
+
+- **`left`** : The value of the left-hand edge of the plot.
+
+- **`right`** : The value of the right-hand edge of the plot.
+
+Returns
+========
+
+A plot object of type `Plot{BoxplotGraphics}`
+
+Author(s)
+==========
+
+- Matthew Lake (Github: https://github.com/mgtlake)
+
+Examples
+=========
+
+    julia> boxplot([1, 2, 3, 7], title = "Test")
+
+                       Test
+     ┌────────────────────────────────────────┐
+     │    │   ┌──┬───────┐              │     │
+     │    ├───┤  │       ├──────────────┤     │
+     │    │   └──┴───────┘              │     │
+     └────────────────────────────────────────┘
+     0                  4.0                   8
+
+See also
+=========
+
+`Plot`, `histogram`, `BoxplotGraphics`
+"""
+function boxplot(
+        data::AbstractVector;
+        labels::AbstractVector{<:AbstractString} = [" " for _ in 1:size(data, 1)],
+        border = :solid,
+        title::AbstractString = "",
+        margin::Int = 3,
+        padding::Int = 1,
+        color::Symbol = :normal,
+        width::Int = 40,
+        left::Number = minimum(map(minimum, data)) - 1,
+        right::Number = maximum(map(maximum, data)) + 1)
+    margin >= 0 || throw(ArgumentError("Margin must be greater than or equal to 0"))
+    length(labels) == length(data) || throw(DimensionMismatch("Wrong number of labels"))
+
+    width = max(width, 10)
+
+    if left > minimum(map(minimum, data)) || right < maximum(map(maximum, data))
+        throw(ArgumentError("Plot range ($left, $right) too restrictive for data range ($(min(data...)), $(max(data...)))"))
+    end
+
+    area = BoxplotGraphics(data[1], width, color = color,
+                           left = left, right = right, labels = labels)
+    for i in 2:length(data)
+        addseries!(area, data[i])
+    end
+
+    new_plot = Plot(area, title = title, margin = margin,
+                   padding = padding, border = border)
+
+    annotate!(new_plot, :bl, string(left))
+    annotate!(new_plot, :b, string((left + right) / 2))
+    annotate!(new_plot, :br, string(right))
+
+    for label in labels
+        annotate!(new_plot, :l, "   ")
+        annotate!(new_plot, :l, label)
+        annotate!(new_plot, :l, "   ")
+    end
+
+    new_plot
+end
+
+"""
+`boxplot!(plot, data; nargs)` → `Plot`
+
+Mutating variant of `boxplot`, in which the first parameter (`plot`) specifies
+the existing plot to draw on.
+
+See `boxplot` for more information.
+"""
+function boxplot!(
+        plot::Plot{<:BoxplotGraphics},
+        data::AbstractVector{<:Number};
+        label = " ",
+        kw...)
+    !isempty(data)|| throw(ArgumentError("Can't append empty array to boxplot"))
+
+    if plot.graphics.left > minimum(data)
+        plot.graphics.left = minimum(data) - 1
+    end
+    if plot.graphics.right < maximum(data)
+        plot.graphics.right = maximum(data) + 1
+    end
+
+    addseries!(plot.graphics, data)
+
+    annotate!(plot, :l, "   ")
```

Comment:
I've changed this to use row indices which is a lot neater.

I was inclined towards labels on the left so that the reader would read them first, but I'm happy to change this, especially if there's a convention I'm not aware of.
What are your thoughts?

Reply:
I thought about this a little. Since barplot and histogram also put the text on the left side, it actually makes more sense to also put it on the left side here. So I changed my mind on this. Lets leave them on the left side